app/components/hearing-form.js

```diff
@@ -0,0 +1,85 @@
+import Component from '@ember/component';
+import { action, computed } from '@ember/object';
+import moment from 'moment';
+
+export default Component.extend({
+
+  // for checking if user clicks submit button without inputting necessary attributes
+  dateMissing: null,
+
+  timeMissing: null,
+
+  locationMissing: null,
+
+  // to check whether hearing model attributes "location" and "date" are populated
+  hearingSubmitted: null,
+
+  // minimum date for calendar
+  currentDate: new Date(),
+
+  // for clearing search results on the geosearch input
+  onClearSearchResult: () => {},
+
+  @computed()
```

Comment:
I'm surprised this syntax still works — decorators for object literal properties were part of an old proposal. Anyways, it's deprecated now, and we should be using ES6 classes (see above) when using decorators.